examples/platform/asset_groups.py

```diff
@@ -0,0 +1,191 @@
+    """Show how to use the preview asset group function to understand the impact of changes such as changing a query"""
+    print("\n\n preview_asset_group_changes(): Coming Soon \n\n")
+
+
+def main():
+    """This script demonstrates how to use Asset Groups in the SDK and common operations to link to related objects.
+
+    This example does not use command line parsing in order to reduce complexity and focus on the SDK functions.
+    Review the Authentication section of the Read the Docs for information about Authentication in the SDK
+    https://carbon-black-cloud-python-sdk.readthedocs.io/en/latest/authentication/
+
+    This is written for clarity of explanation, not perfect coding practices.
+    """
+    # CBCloudAPI is the connection to the cloud.  It holds the credentials for connectivity.
+    # To execute this script, the profile must have an API key with the following permissions.
+    # If you are restricted in the actions you're allowed to perform, expect a 403 response for missing permissions.
+    # Permissions are set on Settings -> API Access -> Access Level and then assigned to an API Key
+    # GM - Group Management - group-management - CREATE, READ, UPDATE, DELETE: For managing asset groups
+    # Device - General Information - device - READ: For getting device information
+    # Policies - Policies - org.policies - READ: For viewing policy information and pre-viewing the impact of changes
+    #    to policy ranking and asset groups.
+
+    # api = CBCloudAPI(profile="YOUR_PROFILE_HERE")
+    api = CBCloudAPI(profile="ASSET_GROUP_DEV_01")
+
+    # to get all asset groups, a static method is available on the AssetGroup class.
+    # This is useful for listing the groups configured in your org
+    all_asset_groups = AssetGroup.get_all_groups(api)
+    for ag in all_asset_groups:
+        print("asset group name: {}, member count: {}".format(ag.name, ag.member_count))
+
+    # Create an asset group.  The only mandatory field is the Asset Group Name.
+    # It can be created without a policy, which enables the use of group for visibility of specific assets
+    # It can be created without a query, which enables manual assignment of assets to the group later
+    new_asset_group_name = "My Example Asset Group"
+    new_asset_group = AssetGroup.create_group(api, new_asset_group_name, "Demonstrating the SDK")
+    print(new_asset_group)
+
+    # Add a query.  All assets that match this criteria will be dynamically added to the group
+    new_asset_group.query = "os.equals:WINDOWS"
+    # Assign a policy.  All assets in the group may have this policy applied.  If an asset is in more than one group,
+    # policy ranking determines which is the effective policy.
+    bottom_rank_policy = None
+    for p in api.select(Policy).all():
+        if bottom_rank_policy is None or p.position > bottom_rank_policy.position:
+            bottom_rank_policy = p
+    new_asset_group.policy = bottom_rank_policy.id
```

Comment:
Why bottom ranked policy? Seems a bit odd as an example script

Reply:
added explanation - least chance to impact prod when tinkering with a script